tpp-app/src/services/SettingsService.js

```diff
@@ -75,6 +75,28 @@ export const POSTUpdatePreferences = async (flow, mood, sleep, cramps, exercise)
     }
 })
 
+/**
+ * Post a single tracking preference update
+ * @param {string} key representing one of the tracking options
+ * @param {boolean} value true or false representing if this option is being tracked
+ * @returns 
+ */
+export const POSTUpdateOnePreference = async (key, value) => new Promise(async(resolve, reject) => {
```

Comment:
I don't think you need try since any error occurring should be caught in the .catch()
Also don't think you need await since you have the .then()

I would change to this: 
```
export const POSTUpdateOnePreference = async (key, value) => new Promise(async(resolve, reject) => {
    console.log(key, value);
    AsyncStorage.setItem(key, JSON.stringify(value))
        .then(() => {
            console.log(`Updated symptom: ${JSON.stringify(key)}`);
            resolve();
        }).catch((e) => {
            console.log(JSON.stringify(e));
            reject("Unable to update symptom");
        });
}) 
```